<a href="https://colab.research.google.com/github/yoyoboom/python-exercise-1-4/blob/master/Python_Exercise_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
import pandas
import requests
import json
import urllib
from urllib.request import Request, urlopen
from google.colab import drive
drive.mount('/content/drive')
df = pd.read_csv(open('drive/My Drive/msas.csv', 'rb'))

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [0]:
df.head()

CBSA                         NAME  POPESTIMATE2017
0  10180                  Abilene, TX           170219
1  10420                    Akron, OH           703505
2  10500                   Albany, GA           151434
3  10540                   Albany, OR           125047
4  10580  Albany-Schenectady-Troy, NY           886188

In [0]:
geoid = "31000US" + df["CBSA"].astype(str)

A = geoid[:len(geoid)//2]
Z = geoid[len(geoid)//2:]

In [0]:
A = ','.join(A)
Z = ','.join(Z)

A
Z

'31000US29420,31000US29460,31000US29540,31000US29620,31000US29700,31000US29740,31000US29820,31000US29940,31000US30020,31000US30140,31000US30300,31000US30340,31000US30460,31000US30620,31000US30700,31000US30780,31000US30860,31000US30980,31000US31020,31000US31080,31000US31140,31000US31180,31000US31340,31000US31420,31000US31460,31000US31540,31000US31700,31000US31740,31000US31860,31000US31900,31000US32580,31000US32780,31000US32820,31000US32900,31000US33100,31000US33140,31000US33220,31000US33260,31000US33340,31000US33460,31000US33540,31000US33660,31000US33700,31000US33740,31000US33780,31000US33860,31000US34060,31000US34100,31000US34580,31000US34620,31000US34740,31000US34820,31000US34900,31000US34940,31000US34980,31000US35100,31000US35300,31000US35380,31000US35620,31000US35660,31000US35840,31000US35980,31000US36100,31000US36140,31000US36220,31000US36260,31000US36420,31000US36500,31000US36540,31000US36740,31000US36780,31000US36980,31000US37100,31000US37340,31000US37460,31000US37620,31000US3786

In [0]:
tableids = ['B01001', 'B13016', 'B13017']
tableidstring = ",".join(tableids)

tableidstring

'B01001,B13016,B13017'

# Question 1
What city has the highest count of the audience you chose for parameter #1?

In [0]:
# create new table 1
tbid1 = ['B19001']
tbidstr1 = ','.join(tbid1)

tbid1
tbidstr1

'B19001'

In [0]:
# first half
requesturl = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tbidstr1, A)
loadedjson = requests.get(requesturl)
parsedjson1 = json.loads(loadedjson.text)

In [0]:
# the rest half
requesturl = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tbidstr1, Z)
loadedjson = requests.get(requesturl)
parsedjson2 = json.loads(loadedjson.text)

#### what avg. hh income we chose? parameter#1
Range from 50k to 74.9k: B19001 

In [0]:
# build data frame w/ country and population
dfhh = pd.DataFrame(columns = ['country', 'population'], index = list(range(1,len(geoid))))

In [0]:
# put first half data in dataframe
for i in list(range(1,len(geoid)//2)):
  dfhh['country'][i] = [geoid[i]]
  dfhh['population'][i] = parsedjson1['data'][geoid[i]]['B19001']['estimate']['B19001011'] + parsedjson1['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
# put rest half data in dataframe
for i in list(range(192,len(geoid))):
  dfhh['country'][i] = [geoid[i]]
  dfhh['population'][i] = parsedjson2['data'][geoid[i]]['B19001']['estimate']['B19001011'] + parsedjson2['data'][geoid[i]]['B19001']['estimate']['B19001012']

In [0]:
print(dfhh.loc[dfhh['population'] == max(dfhh['population'])])

            country   population
249  [31000US35620]  1.01121e+06


In [0]:
print(df.loc[df['CBSA'] == 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


According to the analysis, NY-NJ-PA has the highest count of avg. hh income b/w 50k to 70k 1.01121e+06

# Question 2
What city has the highest count of the audience you chose for parameter #2?

In [0]:
# table 2
tbid2 = ['B01001']
tbidstr2 = ','.join(tbid2)

tbid2
tbidstr2

'B01001'

In [0]:
# first half data
requesturl = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tbidstr2, A)
loadedjson = requests.get(requesturl)
parsedjson3 = json.loads(loadedjson.text)

In [0]:
# rest half data
requesturl = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tbidstr2, Z)
loadedjson = requests.get(requesturl)
parsedjson4 = json.loads(loadedjson.text)

#### what age range we should target? parameter#2
Range from age 18 to 34:
'B01001003', 'B01001004', 'B01001005', 'B01001006', 'B01001007', 'B01001008', 'B01001009', 'B010010010', 'B010010011', 'B010010012'

In [0]:
# age range
age_range = ['B01001007','B01001008','B01001009','B01001010','B01001011','B01001012']

In [0]:
df1 = pd.DataFrame(columns=['location', 'population'], index=list(range(1,len(geoid))))

In [0]:
# put first half data in dataframe
for i in list(range(1,len(geoid)//2)):
    df1['location'][i]=[geoid[i]]
    df1['population'][i] = parsedjson3['data'][geoid[i]]['B01001']['estimate'][age_range[0]] + parsedjson3['data'][geoid[i]]['B01001']['estimate'][age_range[1]] + parsedjson3['data'][geoid[i]]['B01001']['estimate'][age_range[2]] + parsedjson3['data'][geoid[i]]['B01001']['estimate'][age_range[3]] + parsedjson3['data'][geoid[i]]['B01001']['estimate'][age_range[4]] + parsedjson3['data'][geoid[i]]['B01001']['estimate'][age_range[5]] 

In [0]:
# put rest half data in dataframe
for i in list(range(192,len(geoid))):
  df1['location'][i] = [geoid[i]]
  df1['population'][i] = parsedjson4['data'][geoid[i]]['B01001']['estimate'][age_range[0]]+parsedjson4['data'][geoid[i]]['B01001']['estimate'][age_range[1]]+parsedjson4['data'][geoid[i]]['B01001']['estimate'][age_range[2]]+parsedjson4['data'][geoid[i]]['B01001']['estimate'][age_range[3]]+parsedjson4['data'][geoid[i]]['B01001']['estimate'][age_range[4]]+parsedjson4['data'][geoid[i]]['B01001']['estimate'][age_range[5]]

In [0]:
print(df1.loc[df1['population'] == max(df1['population'])])

           location   population
249  [31000US35620]  2.33054e+06


In [0]:
print(df.loc[df['CBSA'] == 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


According to the analysis, NY-NJ-PA has the highest count of age range b/w 18 to 34, 2.33054e+06

# Question 3
What city has the highest count of the audience you chose for parameter #3?

In [0]:
# build new table 3
tbid3 = ['B03003']
tbidstr3 = ','.join(tbid3)

tbid3
tbidstr3

'B03003'

In [0]:
# first half data
requesturl = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tbidstr3, A)
loadedjson = requests.get(requesturl)
parsedjson5 = json.loads(loadedjson.text)

In [0]:
# rest half data
requesturl = 'http://api.censusreporter.org/1.0/data/show/latest?table_ids=%s&geo_ids=%s' % (tbidstr3, Z)
loadedjson = requests.get(requesturl)
parsedjson6 = json.loads(loadedjson.text)

#### Should we look for areas with large Hispanic populations? Parameter #3

In [0]:
df2 = pd.DataFrame(columns=['location', 'population'], index=list(range(1,len(geoid))))

In [0]:
# put first half data in dataframe
for i in list(range(1,len(geoid)//2)):
    df2['location'][i] = [geoid[i]]
    df2['population'][i] = parsedjson5['data'][geoid[i]]['B03003']['estimate']['B03003003']

In [0]:
# put rest half data in dataframe
for i in list(range(192,len(geoid))):
    df2['location'][i] = [geoid[i]]
    df2['population'][i] = parsedjson6['data'][geoid[i]]['B03003']['estimate']['B03003003']

In [0]:
print(df2.loc[df2['population'] == max(df2['population'])])

           location   population
210  [31000US31080]  6.00354e+06


In [0]:
print(df.loc[df['CBSA'] == 31080])

      CBSA                                NAME  POPESTIMATE2017
210  31080  Los Angeles-Long Beach-Anaheim, CA         13353907


According to the analysis, LA-LB-AH has the highest count of Hispanic, 6.00354+06

# Question 4
What city has the highest mean percentage across all three categories?

In [0]:
# create 3 empty sets
hh_income = []
ppl_by_age = []
ppl_by_his = []

In [0]:
# put data in sets
for i, row in dfhh.iterrows():
    per_income = dfhh['population'][i]/max(dfhh['population'])
    per_age = df1['population'][i]/max(df1['population'])
    per_his = df2['population'][i]/max(df2['population'])
    hh_income.append(per_income)
    ppl_by_age.append(per_age)
    ppl_by_his.append(per_his)

In [0]:
dfhh['percentage'] = hh_income
df1['percentage'] = ppl_by_age
df2['percentage'] = ppl_by_his

In [0]:
# percentage mean
Pmean = []
for i, row in dfhh.iterrows():
    Mean = (dfhh['percentage'][i] + df1['percentage'][i] + df2['percentage'][i])/3
    Pmean.append(Mean)
dfhh['percent mean'] = Pmean

In [0]:
print(dfhh.loc[dfhh['percent mean'] == max(dfhh['percent mean'])])

            country   population  percentage  percent mean
249  [31000US35620]  1.01121e+06         1.0      0.940082


In [0]:
print(df.loc[df['CBSA'] == 35620])

      CBSA                                   NAME  POPESTIMATE2017
249  35620  New York-Newark-Jersey City, NY-NJ-PA         20320876


According to the analysis, NY-NJ-PA has the highest mean percent across all 3 categories.